spec/controllers/prison_api/api/offenders_controller_spec.rb

```diff
@@ -138,4 +138,28 @@
       expect(assigns(:offenders).map(&:offenderNo)).to eq(%w[G1006FX G1007FX G1008FX G1009FX G1010FX])
     end
   end
+
+  describe "#assessments" do
+    context "when offenders are missing a category code" do
+      let!(:offender_cat_a) { create(:offender, categoryCode: "A") }
+      let!(:offender_cat_d) { create(:offender, categoryCode: "D") }
+      let!(:offender_no_cat) { create(:offender, categoryCode: nil) }
+
+      render_views
+
+      it "excludes them from the list" do
+        offender_nos = [offender_cat_a, offender_cat_d, offender_no_cat].map(&:offenderNo)
+        post :assessments, body: offender_nos.to_json, format: :json
+        expect(response).to be_successful
+
+        response_array = JSON.parse(response.body)
+        expect(response_array.count).to eq(2)
+        expect(response_array.map { |a| [a["offenderNo"], a["classificationCode"]] })
+          .to eq [
+                   [offender_cat_a.offenderNo, "A"],
+                   [offender_cat_d.offenderNo, "D"],
+                 ]
+      end
```

Comment:
This is probably a stupid question Ollie and will just show my lack of understanding. You are checking the response.body has the correct offenders in the list, could you also have checked @offenders in the controller,  where you have made the code change also is returning what you expect? Or would this be a different type of test?

Reply:
@Katy600 you're absolutely right – I could have checked that the controller sets `@offenders`. Instead, I've used `render_views` to tell the controller to render views, and then I'm inspecting the JSON response to make sure it contains what we expect.

This sort of test is probably more suited towards being a 'request spec' rather than a 'controller spec' – which is more high level, a bit like a feature test for API endpoints – but I've implemented it as a controller spec to be more consistent with the existing tests.

Reply:
Thank you @ollietreend. That makes lots of sense now! :)